core/src/main/java/org/infinispan/commands/CommandsFactory.java

```diff
@@ -125,6 +125,15 @@
    InvalidateCommand buildInvalidateFromL1Command(Address origin, Set<Flag> flags, Collection<Object> keys);
 
    /**
+    * Builds an expired remove command that is used to remove only a specific expired entry
+    * @param key the key of the expired entry
+    * @param value the value of the entry when it was expired
+    * @param lifespan the lifespan that expired from the command
+    * @return a RemovedExpiredCommand
+    */
+   RemoveExpiredCommand buildRemoveExpiredCommand(Object key, Object value, Long lifespan);
```

Comment:
Also, do you really need to add a new command here? Assuming I understood correctly what `RemoveExpiredCommand` does, couldn't you have a built a `ReadWriteCommand` with a function that removes if both the value and expired entry are the same? It should work with minimal work without the need a brand new command :|